### DISCUSSION TITLE
ESP32-C3 generic USB build merged with generic build

### DISCUSSION THREAD

**Discussion by steveh127** [micropython/micropython#12269] [ESP32] [answered]
Until very recently the ESP32 C3 had two builds available one of which was a USB version which was necessary if you wanted to access the REPL via USB. This has now been merged with the generic version and USB access to REPL now works from the generic build so long as you install a recent nightly build.

The v1.20 release however is the old build. Install this and you don't have REPL access via USB. Perhaps the v1.20 USB release could be made available at least until v1.21 is released?

**Comment by robert-hh** [comment]
It seems that the recent change for unified ESP32 builds needs a fix-up with respect to the C3 boards. Those come at least in 2 flavors: 

a)  boards with Generic USB support and REPL at /dev/ttyACMx
b)  boards with REPL over serial and a serial/USB bridge on the board via /dev/ttyUSBx

Building the firmware an deploying them to the board I get a REPL for type a) boards, but not for type b) boards any more. Before that change, there were two different boards listed.

@jimmo Any hints?

Note: At type b) boards there is the additional complications that some boards miss the transistors and require both rts and dtr to be lowered. But that's a hardware trap..

**Reply by jimmo** [reply]
Thanks @robert-hh -- I see what happened here now...

The intent was for both (a) and (b) to support the REPL on both. This requires `MICROPY_HW_ENABLE_UART_REPL`, which was being enabled implicitly on the original `GENERIC_C3` variant. The `GENERIC_C3_USB` variant didn't enable the UART REPL.

So, like the generic S2 and S3 boards, we now need to explicitly enable `MICROPY_HW_ENABLE_UART_REPL` in `mpconfigvariant.h`. I have added a commit to https://github.com/micropython/micropython/pull/12240 to do this.

**Comment by steveh127** [comment]
I only tested with a Seeed XIAO ESP32 C3 board which is one  Roberts type a) boards using /dev/ttyACMx. Would definitely seem to be a good idea to, at least temporarily, restore the C3 USB build if some C3 boards can't use the merged build.

**Comment by jimmo** [comment, marked as answer]
> Perhaps the v1.20 USB release could be made available at least until v1.21 is released?

@steveh127 Yes this is the plan, we just need to apply the relevant renames to the historical firmware builds on the micropython.org web server. (See https://github.com/micropython/micropython/pull/12240#issuecomment-1679882819).

Essentially the result is that the "ESP32_GENERIC_C3" board will have what was previously the GENERIC_C3_USB v1.20 release, and then the new merged version going forward.

This will happen early this week, hopefully tomorrow.
